pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/pdf/DownloadPDF.java

```diff
@@ -84,39 +86,60 @@ public DownloadPDF(@NonNull String requestBody) {
 
     public Optional<byte[]> getPDFs(long userIdRequest, String realm, String requestBody) {
         UserDto user = new UserDao().get(userIdRequest).orElseThrow();
-        Optional<byte[]> pdfBytes = Optional.empty();
+        Optional<byte[]> maybePdf = Optional.empty();
         if (StringUtils.isNotBlank(this.ddpParticipantId)) {
             DDPInstance ddpInstance = DDPInstance.getDDPInstance(realm);
             if (ddpInstance != null && StringUtils.isNotBlank(ddpParticipantId)) {
                 String fileName = "";
                 if (configName == null) {
-                    pdfBytes = getPDFBundle(ddpInstance, requestBody, user);
+                    maybePdf = getPDFBundle(ddpInstance, requestBody, user);
                 } else {
-                    pdfBytes = generateSinglePDF(requestBody, configName, user, ddpInstance);
+                    maybePdf = generateSinglePDF(requestBody, configName, user, ddpInstance);
+                }
+                if (maybePdf.isPresent()) {
+                    byte[] pdfByte = maybePdf.get();
+                    if (pdfByte.length > 0) {
+                        savePDFinBucket(ddpInstance.getName(), ddpParticipantId, new ByteArrayInputStream(pdfByte),
+                                fileName, user.getId());
+                    } else {
+                        throw new RuntimeException("PDF size was zero!" + fileName);
+                    }
                 }
-                pdfBytes.ifPresent(bytes -> {
-                    savePDFinBucket(ddpInstance.getName(), ddpParticipantId, new ByteArrayInputStream(bytes), fileName, user.getId());
-                });
-                pdfBytes.orElseThrow();
+                maybePdf.orElseThrow();
```

Comment:
Recommend you specify the exception to be thrown and add some logging here.

Reply:
Yes I agree, changed it